### PR TITLE
Add support for `decimal128` to `bsonproto`

### LINKED ISSUE #43: Support `decimal128` values in `bsonproto`

### PR DESCRIPTION
**Pull request by AlekSi**
Closes #43.

### CODE REVIEW

File: bsonproto/bsonproto_test.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ func TestScalars(t *testing.T) {
 	}, {
 		v: int64(1234567890123456789),
 		b: []byte{0x15, 0x81, 0xe9, 0x7d, 0xf4, 0x10, 0x22, 0x11},
+	}, {
+		v: Decimal128{H: 2024, L: 404},
```
Comment (cristaloleg) [inline bsonproto/bsonproto_test.go:75]: Deep 😂

**Review by cristaloleg — APPROVED** [review]

File: bsonproto/decimal128.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package bsonproto
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+// Decimal128 represents BSON scalar type decimal128.
+type Decimal128 struct {
+	L uint64
```
[the remark this answers is not in the document]
Comment (cristaloleg) [reply bsonproto/decimal128.go:10]: and 'Hi' respectively

[the remark this answers is not in the document]
Comment (AlekSi) [reply bsonproto/decimal128.go:10]: It is called H and L in both MongoDB Go drivers are checked, so…

File: bsonproto/bsonproto_test.go (hunk 1 above)
Comment (AlekSi) [reply bsonproto/bsonproto_test.go:75]: 2024-04-04 (today), but also 404, yeah

**Pull request merged**